static/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -552,24 +583,49 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
           />
         </Body>
         <Footer>
-          <ButtonBar gap={1}>
+          <StyledButtonBar gap={1}>
             <Button
               external
               href="https://docs.sentry.io/product/dashboards/custom-dashboards/#widget-builder"
             >
               {t('Read the docs')}
             </Button>
-            <Button
-              data-test-id="add-widget"
-              priority="primary"
-              type="button"
-              onClick={this.handleSubmit}
-              disabled={state.loading}
-              busy={state.loading}
-            >
-              {isUpdatingWidget ? t('Update Widget') : t('Add Widget')}
-            </Button>
-          </ButtonBar>
+            <ButtonBar gap={1}>
+              {fromLibrary ? (
+                <Button
+                  data-test-id="add-widget"
+                  type="button"
+                  onClick={() => {
+                    if (dashboard && onAddLibraryWidget) {
+                      openDashboardWidgetLibraryModal({
```

Comment:
Do I need to call `closeModal()` before opening a new one here?

Reply:
Yeah I inspected the HTML that gets rendered, there isn't any weird nesting as far as I can tell, but just wanted to double check that I wasn't missing anything and we didn't keep mounting these modals on the DOM or something!